_posts/2024-01-01-roadmap.md

```diff
@@ -0,0 +1,179 @@
+  
+  *We recommend those using spring-security-keycloak plugin join forces in development and testing of OAuth2 OpenID Connect integration.*
+  
+* **Migrate to Java Enterprise Edition**
+  
+  GeoServer is a Java Web Application comprised of a number of "servlets" that can be run by an application server. The specification of how these components work together is defined by the Java Enterprise Edition specification. This specification is now managed by the Eclipse Foundation as Jakarta Enterprise Edition.
+  
+  With the change to Jakarta Enterprise Edition we expect a number of compatibility issues:
+  
+  * The charts extension is based on [eastwood charts](https://www.jfree.org/eastwood/) last updated in 2008.
+    
+    This library is not compatible with Jakarta Enterprise Edition and will need to be replaced.
+    
+  * mapfish-print-v2  
+  
+    This library is not compatible with Jakarta Enterprise Edition and will need to be updated or replaced. 
+  
+  Application Servers that support Jakarta Enterprise Edition:
+  
+  * Apache Tomcat 10.1 / Jakarta Enterprise Edition 10 / Servlet 6 / Java 17+
+  * Jetty 12.0 / Jakarta Enterprise Edition 10 / Servlet 6 / Java 17+
+  
+  *When ready we will need volunteers to test on the new application servers and update the binary release and documentation to reflect the new environment. Organizations operating in a managed environment may wish to pursue permission to operate Tomcat 10.1 ahead of this planned change.*
+  
+  The spring-framework 6 uses the newer Jakarta Enterprise Edition specification.
+  
+* **Upgrade to Java 17** (Deadline October 2024)
+  
+  GeoServer presently uses Java 11 LTS which reaches end-of-life on RHEL October 2024 (although other vendors provided continued support).
+  
+  With the change to Java 17 we expect a number of libraries we use to require updating or replacing.
+  
+  *GeoServer is presently building on Java 17, however documentation will need to be updated when Java 11 support is dropped. Organizations may wish to pursue permission to operate Java 17 LTS ahead of this planned change.*
+  
+  The spring-framework 6 and Jakarta Enterprise Edition application servers require Java 17 as a minimum.
+  
+* **Migrate to ImageN 1.0**
+  
+  The Java Advanced Imaging library is used as the engine for our image and raster processing capabilities. This library reached end-of-life with the last JAI 1.1.3 release in 2005.
+  
+  This library has received considerable investment from our community with GeoSolutions heading up the JAI-EXT project to better work with geospatial datasets, operations and analysis including recent support for hyperspectral imagery.
+  
+  We have been planning for this migration for some time:
+  
+  1. First working with Boundless to outline the creation of a new library ($150k) after assessing alternatives in the Java ecosystem (nothing matched JAI on-demand capabilities required for geospatial content).
```

Comment:
I did not hear anything about this. Might want to add something like "This was an internal effort that was eventually dropped".

Reply:
I am trying to put dollar figures on the activity to indicate the sense of scale; I was doing the experiment at LocationTech so perhaps you did not notice at the time; although I think I shared the assessment of different image processing libraries.

Reply:
I would still clarify the effort was abandoned and why.

Reply:
I have provided more background on the creation of the ImageN project.